include/envoy/stats/stats.h

```diff
@@ -83,17 +135,29 @@ typedef std::unique_ptr<Timespan> TimespanPtr;
 /**
  * A timer that can capture timespans.
  */
-class Timer {
+class Timer : public Metric {
 public:
   virtual ~Timer() {}
 
   virtual TimespanPtr allocateSpan() PURE;
-  virtual std::string name() PURE;
+  virtual void recordDuration(std::chrono::milliseconds ms) PURE;
 };
 
 typedef std::shared_ptr<Timer> TimerSharedPtr;
 
 /**
+ * A histogram that captures values one at a time.
+ */
+class Histogram : public Metric {
```

Comment:
Question: Is it worth creating a full on histogram interface vs. just having Timer take some kind of flag as an indication to the stats sink? They are basically the same thing.